internal/faults/common.go

```diff
@@ -0,0 +1,278 @@
+// Package faults provides common headers and client-side fault injection
+// functionality.
+package faults
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"math/rand/v2"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/reddit/baseplate.go/internal/prometheusbpint"
+)
+
+const (
+	promNamespace      = "faultbp"
+	clientNameLabel    = "fault_client_name"
+	serviceLabel       = "fault_service"
+	methodLabel        = "fault_method"
+	protocolLabel      = "fault_protocol"
+	successLabel       = "fault_success"
+	delayInjectedLabel = "fault_injected_delay"
+	abortInjectedLabel = "fault_injected_abort"
+)
+
+var (
+	TotalRequests = promauto.With(prometheusbpint.GlobalRegistry).NewCounterVec(prometheus.CounterOpts{
+		Namespace: promNamespace,
+		Name:      "fault_requests_total",
```

Comment:
for new code we don't recommend using `Namespace` or `Subsystem`, as that makes it much harder to search for the code emitting a metric, just put `faultbp_fault_requests_total` in the `Name` instead.

Reply:
Done!